### PR TITLE
EVG-14156 Add task will run, will not run and blocked statuses

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-14156](https://jira.mongodb.org/browse/EVG-14156)

### Description 
This PR adds a task will run status and a task will not run status. These statuses replace undispatched. 
It also adds blocked as a display status so we can filter and sort on it. 

There will be an upcoming spruce pr. To support this.

### CODE REVIEW

File: model/task/db.go [hunk 1]
```diff
@@ -194,6 +194,42 @@ var (
...
+				// A task will run if it is activated and does not have any blocking deps
+				{
+					"case": bson.M{
+						"$and": []bson.M{
+							{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
+							{"$eq": []interface{}{"$" + ActivatedKey, true}},
+							{
+								"$or": []bson.M{
+									{DependsOnKey: 0},
+									{"$eq": []interface{}{"$" + bsonutil.GetDottedKeyName(DependsOnKey, DependencyUnattainableKey), false}},
```
Comment (ablack12) [inline model/task/db.go:206]: Could also do $neq "true" to shorten this up

**Review by ablack12 — COMMENTED** [review]
will Dispatched still be its own status then? Or should this be lumped into "will run"?

**Review by malikchaya2 — COMMENTED** [review]
Will "aborted" still be handled the same way?
How should the main patch status behave when there is a "will run" or "will not run" task in it?

File: model/task/db.go [hunk 2]
```diff
@@ -194,6 +194,42 @@ var (
 					},
 					"then": evergreen.TaskTimedOut,
 				},
+				// A task will run if it is activated and does not have any blocking deps
```
Comment (SupaJoon) [inline model/task/db.go:197]: it looks like calling `FindTasksByVersion` will execute this code path  to calculate the displayStatus when [this code is ran](https://github.com/evergreen-ci/evergreen/blob/main/model/task/task.go#L2770) . We may also be able to utilize this code path for the task.FindByIdExecution (called in the Task resolver) and potentially remove the updatedStatus calculation logic in taskResolver.Status

Comment (khelif96) [reply model/task/db.go:197]: Good catch just added this

**Comment by khelif96** [comment]
> will Dispatched still be its own status then? Or should this be lumped into "will run"?

Dispatched will remain as its own status these only override the undispatched task statuses

**Comment by khelif96** [comment]
> Will "aborted" still be handled the same way?
> How should the main patch status behave when there is a "will run" or "will not run" task in it?

Aborted shouldn't be affected. Only difference is it can now be used for sorting and filtering
Patch status will not change if all the tasks are either `will run` or `will not run` the patch status should remain `undispatched` since `will run` and `will not run` are subsets of undispatched.

File: model/task/db.go (hunk 1 above)
Comment (khelif96) [reply model/task/db.go]: `$ne` *

**Review by malikchaya2 — APPROVED** [review]

File: model/task/db.go [hunk 3]
```diff
@@ -818,6 +854,53 @@ func FindOneIdAndExecution(id string, execution int) (*Task, error) {
...
+		addDisplayStatus,
+	}
+	if err := db.Aggregate(OldCollection, pipeline, &tasks); err != nil {
+		return nil, errors.Wrap(err, "Couldn't find task")
+	}
+	if len(tasks) != 0 {
+		t := tasks[0]
+		return &t, nil
+	}
+	return nil, errors.New("Cant find task")
```
Comment (ablack12) [inline model/task/db.go:901]: incredibly tiny nit: shouldn't capitalize error messages

File: model/task/task.go [hunk 4]
```diff
@@ -1370,13 +1370,28 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 
 }
 
+// This should reflect the addDisplayStatus aggregation step
```
Comment (ablack12) [inline model/task/task.go:1373]: Nit: can you use the function name instead of This so it's official documentation?

**Review by ablack12 — APPROVED** [review]
LGTM with nits

**Pull request merged**